articles/azure-app-configuration/enable-dynamic-configuration-dotnet-core-push-refresh.md

```diff
@@ -175,6 +175,74 @@ A random delay is added before the cached value is marked as dirty to reduce pot
+    public string EventType;
+}
+```
+This class stores the `ResourceUri` of the resource which triggered the `PushNotification`, the `Synchronization Token` to be added to the next request to the App Configuration Service, and the Type of Event which triggered the `PushNotification`.
+
+**TryCreatePushNotification**
+```cs
+EventGridEvent.TryCreatePushNotification(out PushNotification pushNotification)
+```
+This method uses the `Data`, `EventType`, and `Subject` stored in the Event Grid Event to try and create a `PushNotification`. If successful returns *true* and an out `PushNotification` parameter. Otherwise the parameter will be null and the method will return *false*.
+
+**ProcessPushNotification**
+```cs
+ProcessPushNotification(PushNotification pushNotification, TimeSpan? maxDelay)
+```
+This method is called by the refresher and provider to process the `PushNotification`. `ProcessPushNotification()` will invoke `UpdateSyncToken()` and then invoke `SetDirty()` to ensure the most recent configuration is provided to the user.
+
+**Implementation**
+To use a `sync-token` in your push refresh workflow, make the following changes to your **Program.cs** file:
+
+1. Add a dependency on Azure.messaging.EventGrid Package:
+```cs
+using Azure.Messaging.EventGrid;
+```
+2. Make the following changes to the RegisterRefreshEventHandler method:
+```cs
+ private static void RegisterRefreshEventHandler()
+        {
+            string serviceBusConnectionString = Environment.GetEnvironmentVariable(ServiceBusConnectionStringEnvVarName);
+            string serviceBusTopic = Environment.GetEnvironmentVariable(ServiceBusTopicEnvVarName);
+            string serviceBusSubscription = Environment.GetEnvironmentVariable(ServiceBusSubscriptionEnvVarName);
+            SubscriptionClient serviceBusClient = new SubscriptionClient(serviceBusConnectionString, serviceBusTopic, serviceBusSubscription);
+            serviceBusClient.RegisterMessageHandler(
+                handler: (message, cancellationToken) =>
+                {
+                    EventGridEvent eventGridEvent = EventGridEvent.Parse(BinaryData.FromBytes(message.Body));
+
+                    eventGridEvent.TryCreatePushNotification(out PushNotification pushNotification);
+
+                    _refresher.ProcessPushNotification(pushNotification);
+
+                    return Task.CompletedTask;
+                },
+                exceptionReceivedHandler: (exceptionargs) =>
+                {
+                    Console.WriteLine($"{exceptionargs.Exception}");
+                    return Task.CompletedTask;
+                });
+        }
+```
+
+The `SetDirty` method is called in the PushModel. `SetDirty()` is invoked internally in the `ProcessPushNotification` method to cache values for key-values registered for refresh as dirty. This ensures that the next call to `RefreshAsync` or `TryRefreshAsync` re-validates the cached values with App Configuration and updates them if needed.
```

Comment:
This is already explained in the previous section. Suggest to change it to:

"Notice that you don't need to invoke the `SetDirty` method anymore. This is because `SetDirty` is always invoked by the `ProcessPushNotification` method internally. The cached value will be marked as dirty after a random delay of up to 30 seconds, but this max delay can be overridden by passing an optional `TimeSpan` parameter to the `ProcessPushNotification` method."